src/vs/workbench/services/search/node/fileSearch.ts

```diff
@@ -628,16 +628,18 @@ export class Engine implements ISearchEngine<IRawFileMatch> {
 	private folderQueries: IFolderQuery[];
 	private extraFiles: URI[];
 	private walker: FileWalker;
+	private numThreads?: number;
 
-	constructor(config: IFileQuery) {
+	constructor(config: IFileQuery, numThreadsPromise?: number) {
```

Comment:
You can probably call this `numThreads` instead of `numThreadsPromise`, since it's not a promise